tests/basilisp/test_core_fns.lpy

```diff
@@ -1382,6 +1407,42 @@
     {:a "A" 'b 2}                                {:a "a" 'b 2}
     ^{:good :yes :great :also-yes} {:a "A" 'b 2} ^{:good :yes :great :also-yes} {:a "a" 'b 2}))
 
+(deftest keys-test
+  (are [expected m] (= expected (set (keys m)))
+    #{}      {}
+    #{}      nil
+    #{:a}    {:a 1}
+    #{:a}    {:a nil}
+    #{:a :b} {:a 1 :b 2}
+    #{"a"}   {"a" 1}
+    #{}      (python/dict {})
+    #{:a}    (python/dict {:a 1})))
```

Comment:
FWIW you could also do `#py {}` for these.

Reply:
Well that much is true. Bizarre that the non-literal version isn't causing issues, though.

Anyway, this wasn't important so no worries.

Reply:
Thinking about this more I'm fairly sure it's going to be a ~problem with~ limitation of `basilisp.test/are` and templating, not the reader.

`(python/dict ...)` is a list which is hashable (to `are`) but `#py {}` is a `dict` which is not hashable.

Reply:
The templates used by `are` places the expression elements in a hash map and then uses that to `postwalk-replace` the actual expressions. That calls `(contains? replacements expr)` which in our case `expr` is `#py {}`.

The underlying mapping library looks up items by hash, but dicts are not hashable so it throws. Nothing is wrong with the reader or macros.

Reply:
I'm not strongly convinced that's necessary. `postwalk-replace` is built on `postwalk` and users are free to define the variant you've provided as they wish.

I think swallowing that exception is more likely to suppress errors than it is to be helpful.